docs/current_docs/manuals/user/remotes/remote-repositories.mdx

```diff
@@ -11,3 +11,91 @@ Here is an example of a Go builder Dagger Function that accepts a remote Git add
 ```shell
 dagger -m github.com/kpenfound/dagger-modules/golang@v0.2.0 call build --source=https://github.com/dagger/dagger --args=./cmd/dagger export --path=./build
 ```
+
+## Dagger Module Ref Schemes
+
+Dagger supports various reference ("ref") schemes for remote modules, accommodating both HTTP and SSH protocols. These schemes are compatible with all Git repositories hosted on platforms such as GitHub, GitLab, BitBucket, Azure DevOps, Codeberg, and Sourcehut.
```

Comment:
On reading the entire page, I felt like there was something missing. This page is about remote repositories. In the first section, we describe how to access a public (non-module) repository. In this section, we describe how to access a private module repository.  But this section is silent about how to access a private non-module repository. Can we add an example for that as well?

My proposal is, let's split this page into two sub-sections: (i) access public repositories (already documented) and (ii) accessing private repositories. The private Dagger module repository case which is described here then becomes a subset of (ii). Let me know if this makes sense or if I misunderstood something.

Reply:
> My proposal is, let's split this page into two sub-sections: (i) access public repositories (already documented) and (ii) accessing private repositories.

I'm not sure about splitting the page in two. However, I do agree with you that I feel the message is not clearly communicated. What I feel it's happening is that the first section talking about public repositories seems to be unnecessarily long and deviates the attention to the core idea of the page.  My take on this one is to reword the first part and make it more "private repositories" focus. 

Having said that, if you still prefer to go with the split, I'm ok with giving that approach a try :pray:

Reply:
@vikram-dagger not sure to understand `private non-module repository`, do you mean: passing a private repo as a directory in a function call?

Reply:
Adding a commit trying that out as is, so that we can just remove the commit

Reply:
To follow-up with above: the only issue I see with the split is that there is "shared" context about refs, and in reality, it is highly probable that we will handle with a more precise ref style for function arguments

But, currently, function arguments and modules ref do differ. SSH_AUTH_SOCK is indeed supported in both cases, just function arguments still follow the BuildKit ref style